src/main/java/net/htmlcsjs/htmlTech/common/laserpipe/LaserPipeRenderer.java

```diff
@@ -1,267 +1,52 @@
+import gregtech.api.pipenet.block.BlockPipe;
+import gregtech.api.pipenet.block.IPipeType;
 import gregtech.api.pipenet.tile.IPipeTile;
-import gregtech.api.util.GTUtility;
-import gregtech.api.util.ModCompatibility;
+import gregtech.api.unification.material.Material;
+import gregtech.client.renderer.pipe.PipeRenderer;
 import net.htmlcsjs.htmlTech.HtmlTech;
-import net.htmlcsjs.htmlTech.common.laserpipe.tile.TileEntityLaserPipe;
-import net.minecraft.block.state.IBlockState;
-import net.minecraft.client.renderer.BufferBuilder;
-import net.minecraft.client.renderer.GlStateManager;
-import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
-import net.minecraft.client.renderer.block.model.ModelResourceLocation;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.client.renderer.texture.TextureMap;
-import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
-import net.minecraft.item.ItemStack;
-import net.minecraft.util.BlockRenderLayer;
-import net.minecraft.util.EnumBlockRenderType;
-import net.minecraft.util.EnumFacing;
 import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.Vec3d;
-import net.minecraft.world.IBlockAccess;
-import net.minecraftforge.client.MinecraftForgeClient;
-import net.minecraftforge.client.event.ModelBakeEvent;
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.common.model.IModelState;
-import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
-import org.apache.commons.lang3.ArrayUtils;
-import org.apache.commons.lang3.tuple.Pair;
 
+import javax.annotation.Nullable;
 import java.util.HashMap;
 import java.util.Map;
 
-public class LaserPipeRenderer implements ICCBlockRenderer, IItemRenderer {
+public class LaserPipeRenderer extends PipeRenderer {
 
-    public static final ModelResourceLocation MODEL_LOCATION = new ModelResourceLocation(new ResourceLocation(GTValues.MODID, "laser_pipe_normal"), "normal");
     public static final LaserPipeRenderer INSTANCE = new LaserPipeRenderer();
-    public static EnumBlockRenderType BLOCK_RENDER_TYPE;
-    private static final ThreadLocal<BlockRenderer.BlockFace> blockFaces = ThreadLocal.withInitial(BlockRenderer.BlockFace::new);
-    private final Map<LaserPipeType, PipeTextureInfo> pipeTextures = new HashMap();
+    private final Map<LaserPipeType, TextureAtlasSprite> pipeTextures = new HashMap();
 
-    public static void preInit() {
-        HtmlTech.logger.info("initializing laser pipe renderer");
-        BLOCK_RENDER_TYPE = BlockRenderingRegistry.createRenderType("ht_laser_pipe");
-        BlockRenderingRegistry.registerRenderer(BLOCK_RENDER_TYPE, INSTANCE);
-        MinecraftForge.EVENT_BUS.register(INSTANCE);
-        TextureUtils.addIconRegister(INSTANCE::registerIcons);
+    public LaserPipeRenderer() {
+        super("ht_laser_pipe", new ResourceLocation(HtmlTech.MODID, "laser_pipe_normal"));
```

Comment:
Could you move this to `HTTextures.java`?

Reply:
wdym? What exactly should be put there?

Reply:
Wdym? what exactly should be put there

Reply:
I ment, can you move the `resourceLocation` to `HTTextures.java`